frontend_tests/node_tests/message_fetch.js

```diff
@@ -347,6 +347,13 @@ run_test('loading_newer', () => {
         });
 
         assert.equal(msg_list.data.fetch_status.can_load_newer_messages(), true);
+
+        msg_list = simulate_narrow();
+        msg_list.data.fetch_status.get_expected_max_message_id = () => 445;
+        msg_list.data.last = () => ({ id: 444 });
+        data.resp.found_newest = true;
+
+        assert.equal(msg_list.data.fetch_status.can_load_newer_messages(), true);
```

Comment:
I don't understand this new test code; can you explain it?  The last line seems to not do much.

Reply:
oh oops, it didn't call the `test_happy_path` function previously, so the test didn't do anything.
I added explanation for this in the comments.

Reply:
@ryanreh99 Is there a way to structure this test so that if we were to remove `test_happy_path` again, then the assertion `assert.equal(msg_list.data.fetch_status.can_load_newer_messages(), true);` would fail?

In general, an assertion is very weak (and sometimes confusing) if it's testing something to be `true` that was already `true` in the first place.  (There exception to the rule is where you are specifically testing that a function **doesn't** have side effects.)

Reply:
Thanks for the info!

The test fails on removing the `update_expected_max_message_id` line.
But I couldn't find any way to fail an assert on removing the `test_happy_path` line.
and can replace this with a `finish_newer_batch()` line.
I understand the above point.
This test was added only to test the `finish_newer_batch` function which is called by `test_happy_path`. So I left it like this.

Reply:
The second commit resolves this problem.